pkg/ee/validation/machine/providers.go

```diff
@@ -277,19 +277,35 @@ func getKubeVirtResourceRequirements(ctx context.Context, client ctrlruntimeclie
 	}
 
 	var cpuReq, memReq resource.Quantity
-	// if flavor is set, then take the resource details from the vmi preset, otherwise take it from the config
-	if len(flavor) != 0 {
+	// If instancetype is set, memory and cpu come from it.
+	// instancetype will have priority over flavor.
+	switch {
+	case rawConfig.VirtualMachine.Instancetype != nil && len(rawConfig.VirtualMachine.Instancetype.Name) != 0:
+		kubeconfig, err := configVarResolver.GetConfigVarStringValueOrEnv(rawConfig.Auth.Kubeconfig, envKubeVirtKubeConfig)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get KubeVirt kubeconfig from machine config, error: %w", err)
+		}
+		capacity, err := kubevirt.DescribeInstanceType(ctx, kubeconfig, rawConfig.VirtualMachine.Instancetype)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get KubeVirt VMI instancetype: %w", err)
+		}
+		cpuReq = *capacity.CPUCores
+		memReq = *capacity.Memory
+	case len(flavor) != 0:
+		// if flavor is set, then take the resource details from the vmi preset, otherwise take it from the config
```

Comment:
I see that you already added a comment before `switch` statement, please expand the comment and remove this one after flavor.

If code needs some explanation because of its complexity then let's keep it one place (like before switch). It's more readable.

Reply:
Could you remove comment from line 317 as well? The code is self explanatory.

> // parse the KubeVirt resource requests

Reply:
not mine though, removing it.